### PR TITLE
MAINT: stats.gaussian_kde: raise informative message with degenerate data

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
gh-10205

#### What does this implement/fix?
gh-10205 reported that `gaussian_kde` raised a confusing error for their data. The data is degenerate: it lies in a lower-dimensional subspace of the space in which it is expressed. This PR documents that `gaussian_kde` does not support such data and attempts to clarifies the error message and outline a possible solution.

#### Additional information
After merging this, we can remove the defect label from gh-10205 and use it to track two enhancements for KDE: support for singular matrices and eliminating regular use of the covariance matrix inverse (gh-5087). I think we can accomplish both with `scipy.linalg.ldl`.

### CODE REVIEW

**Comment by mdhaber** [comment]
Looks like I need to add the error in a second place or maybe at a higher level.

**Review by rkern — APPROVED** [review]
LGTM! Thanks!

**Comment by rkern** [comment]
Except for the test failure, yeah.

File: scipy/stats/tests/test_kdeoth.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-from scipy import stats
+from scipy import stats, linalg
```
Comment (EngNashar) [inline scipy/stats/tests/test_kdeoth.py:1]: ModuleNotFoundError: No module named 'scipy' the same error after all try

Comment (rkern) [reply scipy/stats/tests/test_kdeoth.py:1]: I'm not sure what you are trying to say, but I think it has more to do with how you tried to build or install scipy than the changes in this PR.

**Comment by mdhaber** [comment]
Thanks @rkern. Looks like those are resolved and the two CI failures are unrelated and being worked on. Please merge if it looks good to you, or let me know if you want someone else to review.

**Review by rkern — APPROVED** [review]

**Pull request merged**